File: my_cpp_pkg/src/add_two_ints_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

class AddTwoIntsClientNode : public rclcpp::Node
{
public:
    AddTwoIntsClientNode() : Node("add_two_ints_client")
    {
        m_client = this->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
    }

    void SendRequest(int64_t a, int64_t b)
    {
        while (!m_client->wait_for_service(std::chrono::seconds(1))) {
            RCLCPP_WARN(this->get_logger(), "Service not available, waiting again...");
        }

        auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
        request->a = a;
        request->b = b;

        m_client->async_send_request(request,
             std::bind(&AddTwoIntsClientNode::CallbackResponse, this, std::placeholders::_1));
    }
private:
    rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedPtr m_client;
    void CallbackResponse(rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedFuture future)
    {
        RCLCPP_INFO(this->get_logger(), "Result: %ld", future.get()->sum);
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<AddTwoIntsClientNode>();
    node->SendRequest(5, 3);
    node->SendRequest(10, 20);
    node->SendRequest(100, 200);
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/add_two_ints_client.py
#!/usr/bin/env python3
from functools import partial
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts


class AddTwoIntsClientNode(Node):
    def __init__(self):
        super().__init__("add_two_ints_client")
        self._client = self.create_client(AddTwoInts, "add_two_ints")

    def call_add_two_ints(self, a: int, b: int) -> None:
        if not self._client.wait_for_service(timeout_sec=1.0):
            self.get_logger().error("Service not available")
            return

        request = AddTwoInts.Request()
        request.a = a
        request.b = b
        future = self._client.call_async(request)
        future.add_done_callback(partial(self.callback_response, request=request))

    def callback_response(self, future, request) -> None:
        try:
            response = future.result()
        except Exception as e:
            self.get_logger().error(f"Service call failed: {e}")
        else:
            self.get_logger().info(
                f"Result of add_two_ints: a={request.a}, b={request.b}, sum={response.sum}"
            )


def main(args=None):
    rclpy.init(args=args)
    node = AddTwoIntsClientNode()
    node.call_add_two_ints(3, 7)
    node.call_add_two_ints(4, 5)
    node.call_add_two_ints(34, 56)
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
